sortingHat/sortingHat.py

```diff
@@ -0,0 +1,347 @@
+import csv
+
+import random
+
+"""
+Global variables here, not great but it works for now. Next step would be to
+populate these through form input so that the instructor can change these as
+we've planned. The skills list should also be handled differently at some
+point, probably something else the instructor should populate.
+"""
+LEARN_WEIGHT = 0.2
+KNOWN_WEIGHT = 0.1
+MIN_SIZE = 4
+MAX_SIZE = 6
+OPT_SIZE = 5
+
+SKILLZ = ["VG", "DB", "ES", "WA", "MA", "UI", "ST", "NW", "SC", "ML", "RB", "CV", "AL", "AI"]
+
+class Lead(Enum):
+    NO_PREF = 0
+    STRONG_LEAD = 1
+    PREFER_LEAD = 2
+    PREFER_FOLLOW = 3
+    STRONG_FOLLOW = 4
+"""
+Global groups are just a stand-in for the database connection
+"""
+students = []
+group_names = ["A","B","C","D","E","F","G","H","I","J","K","L","M","P","R"]
+groups = []
+
+class dataSet:
+    """
+    Class to store csv formatted data from spreadsheet
+    """
+    def __init__(self, path):
+        with open(path, "r") as myfile:
+            self.readData = myfile.readlines()[1:]
+
+##! Object to store student data, once connected to the db this will be a
+##! document schema
+class Student:
+    def __init__(self):
+        self.name = ""
+        self.identikey = ""
+        self.preferences = []
+        self.known_skills = []
+        self.skills_to_learn = []
+        self.desired_leadership = ""
+
+##! Function that returns a formatted string representation of the class for
+##! printing/debugging purposes
+    def __str__(self):
+#        ret_str = "Name: {self.name}\nIdentikey: {self.identikey}\nPreferences\n: {self.preferences}\nKnown skills: {self.known_skills}\nSkills I want to learn: {self.skills_to_learn}\nLeadership: {self.desired_leadership}".format(self=self)
+        ret_str = self.name
+        return ret_str
+##! These let the class be used as a dictionary key 
+    def __hash__(self):
+        return hash((self.name, self.identikey))
+
+    def __eq__(self, other):
+        return (self.name, self.identikey) == (other.name, other.identikey)
+
+    def __ne__(self, other):
+    # Not strictly necessary, but to
+    # avoid having both x==y and x!=y
+    # True at the same time
+        return not(self == other)
+
+
+##! Same thing for groups
+class Group:
+    def __init__(self):
+        self.name = ""
+        self.ident = ""
+        self.members = []
+        self.preferences = {}
+        self.skills_needed = []
+    def __hash__(self):
+        return hash((self.name))
+    def __eq__(self, other):
+        return (self.name) == (other.name)
+    def __ne__(self, other):
+        return not(self == other)
+    def __str__(self):
+#        ret_str = "Group name:\
+#            {self.name}\nPreferences:{self.preferences}\n".format(self=self)
+        ret_str = self.name
+        return ret_str
+
+def createGraph():
+    print("Creating graph")
+    G = nx.Graph()
+    G.add_nodes_from(students, bipartite=0)
+    G.add_nodes_from(groups, bipartite=1)
+    print(G.number_of_nodes())
```

Comment:
I think if we leave the print statements in here, we should add some more formatting/data to them (i.e. print('Number of nodes: ', G....)